### PR TITLE
IPU:for rhel 8 to rhel 9, add function of installing leapp tool from compose or upstream

### PR DESCRIPTION
**Pull request by MiriamDeng**
ID:2159931
The script supports installing leapp tool from compose or upstream.
Signed-off-by: MiriamDeng <mdeng@redhat.com>

### CODE REVIEW

**Comment by MiriamDeng** [comment]
(2/2) Host_RHEL.m9.u2.ovmf.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.8.8.0.x86_64.io-github-autotest-qemu.in_place_upgrade.without_rhsm_rhel8_to_rhel9.q35: STARTED
 (2/2) Host_RHEL.m9.u2.ovmf.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.8.8.0.x86_64.io-github-autotest-qemu.in_place_upgrade.without_rhsm_rhel8_to_rhel9.q35: PASS (4840.00 s)

**Comment by MiriamDeng** [comment]
Compose:
(2/2) Host_RHEL.m9.u2.ovmf.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.8.8.0.x86_64.io-github-autotest-qemu.in_place_upgrade.without_rhsm_rhel8_to_rhel9.compose_install.q35: STARTED
 (2/2) Host_RHEL.m9.u2.ovmf.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.8.8.0.x86_64.io-github-autotest-qemu.in_place_upgrade.without_rhsm_rhel8_to_rhel9.compose_install.q35: PASS (4904.57 s)

**Comment by MiriamDeng** [comment]
Upstream install:
 (2/2) Host_RHEL.m9.u2.ovmf.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.8.8.0.x86_64.io-github-autotest-qemu.in_place_upgrade.without_rhsm_rhel8_to_rhel9.upstream_install.q35: STARTED
 (2/2) Host_RHEL.m9.u2.ovmf.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.8.8.0.x86_64.io-github-autotest-qemu.in_place_upgrade.without_rhsm_rhel8_to_rhel9.upstream_install.q35: PASS (4469.75 s)

**Comment by MiriamDeng** [comment]
@PaulYuuu @fbq815 could you help to review this patch ? Thanks !

File: qemu/tests/in_place_upgrade.py [hunk 1]
```diff
@@ -47,11 +47,12 @@ def run(test, params, env):
         # create an assistant user
         upgrade_test.create_ipuser(test)
         # prepare ipu test env and execute leapp tool
-        upgrade_test.run_guest_cmd(params.get("repo_leapp"))
-        upgrade_test.run_guest_cmd(params.get("ins_leapp_cmd"))
-        upgrade_test.run_guest_cmd(params.get("prepare_env"))
-        upgrade_test.run_guest_cmd(params.get("get_answer_files_source"))
-        upgrade_test.run_guest_cmd(params.get("get_answer_files"))
+        if params.get("com_install") != "yes":
```
Comment (PaulYuuu) [inline qemu/tests/in_place_upgrade.py:50]:
```suggestion
        if params.get_boolean("com_install"):
```

Comment (MiriamDeng) [reply qemu/tests/in_place_upgrade.py]: Thanks ! Updated Please help to review it again !

**Comment by MiriamDeng** [comment]
(2/2) Host_RHEL.m8.u8.product_rhel.qcow2.virtio_blk.up.virtio_net.Guest.RHEL.8.8.0.x86_64.io-github-autotest-qemu.in_place_upgrade.without_rhsm_rhel8_to_rhel9.compose_install.q35: STARTED
 (2/2) Host_RHEL.m8.u8.product_rhel.qcow2.virtio_blk.up.virtio_net.Guest.RHEL.8.8.0.x86_64.io-github-autotest-qemu.in_place_upgrade.without_rhsm_rhel8_to_rhel9.compose_install.q35: PASS (6418.05 s)

**Review by fbq815 — DISMISSED** [review]
LGTM

**Comment by MiriamDeng** [comment]
Update both commit title and patch tile, please help to review it again ！ Thanks @PaulYuuu

**Review by PaulYuuu — APPROVED** [review]

**Pull request merged**